pkg/kubectl/resource_printer.go

```diff
@@ -255,6 +255,12 @@ type JSONPrinter struct {
 
 // PrintObj is an implementation of ResourcePrinter.PrintObj which simply writes the object to the Writer.
 func (p *JSONPrinter) PrintObj(obj runtime.Object, w io.Writer) error {
+	switch obj := obj.(type) {
+	case *runtime.Unknown:
```

Comment:
`runtime.Unknown` didn't `Marshal` correctly?  That would concern me.  @smarterclayton expected?